topostats/processing.py

```diff
@@ -877,7 +898,7 @@ def run_splining(
             )
             return splined_image_data, grainstats_df, splining_molstats
 
-    return None, grainstats_df, None
+    return None, grainstats_df, molstats_df
```

Comment:
If this line is ever reached, which will be the case if `splining_config["run"] == False` then `molstats_df` is whatever has been passed into the function as an argument `molstats_df`. The default for this is `None` and as we've found having `None` is being deprecated in `pd.concat()`.

I've not dug deeply into this but will `molstats_df` always be non-`None` when processing?

Should this perhaps be...

```suggestion
    if molstats_df is not None:
        return None, grainstats_df, molstats_df
    return None, grainstats_df, create_empty_dataframe("splinning_statistics", index="molecule_number")
```

...and the `topostats.utils.COLUMN_SETS` needs another key/value pair for the expected columns?

Reply:
I've stopped it from defaulting to None and instead have the default values similar to `create_empty_dataframe("molecule_statistics", index="molecule_number")`

Reply:
Good spot, I forgot about this. I've now added this in at the start so it's clear what happens to the None's